src/txTransformers.ts

```diff
@@ -96,14 +97,26 @@ const transformTxOutput = (output: TransactionOutput): TransactionOutput => {
   }
 }
 
+/**
+ * If the auxiliary data hash does not match the provided auxiliary data, it is replaced with the
+ * correct hash. This is useful if the auxiliary data was reserialized canonically.
+ * Some clients don't pass in auxiliary data along with the auxiliary data hash. Thus if auxiliary
+ * data is not provided, return the original hash (otherwise we would replace the provided hash
+ * with a hash of undefined data).
+ */
 const transformAuxiliaryDataHash = (
-  auxiliaryData: unknown | undefined,
+  auxiliaryDataHash:
+    | FixlenBuffer<typeof AUXILIARY_DATA_HASH_LENGTH>
+    | undefined,
+  auxiliaryData: Unparsed,
 ): FixlenBuffer<typeof AUXILIARY_DATA_HASH_LENGTH> | undefined =>
-  auxiliaryData ? blake2b256(encodeToCbor(auxiliaryData)) : undefined
+  auxiliaryData === null
```

Comment:
`=== null` or `== null`? Can `auxiliaryData` be also `undefined`? If yes, then `=== null` wouldn't match it.

Reply:
I don't think it should ever be undefined - CBORs I've inspected use `f6` byte for no aux data, which is parsed as `null` by the CBOR lib we use.

Reply:
Also, if it ever was undefined, the `otherItems.filter((item) => item !== undefined)` line in `txParsers.ts` would make a mess anyway :D

Reply:
Wouldn't it be safer to use `==` anyways?

Reply:
I guess it doesn't matter, though it might be confusing for the reader (why do they do `== null` when it should never be `undefined`?). I can change it and mention it in the comment.